plugins/modules/cloud/docker/docker_stack_info.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2018 Dario Zanzico (git@dariozanzico.com)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'status': ['preview'],
+    'supported_by': 'community',
+    'metadata_version': '1.1'
+}
+
+DOCUMENTATION = '''
+---
+module: docker_stack_info
+author: "Jose Angel Munoz (@imjoseangel)"
+short_description: docker stack module info
+description:
+  - Manage docker stacks using the 'docker stack' command
+    on the target node (see examples).
+version_added: "2.11"
+options:
+  name:
+    description:
+      - Stack name
+    type: str
+  action:
+    description:
+      - If set to C(tasks) displays the list of stacks using the C(docker stack ls) command.
+        If not set, the module use "list" by default.
+    type: str
+    choices: ["list", "tasks"]
```

Comment:
I'm not sure whether I like this interface. It looks like the module can do different or even unrelated things, and return them as the same result variable.

Also I don't really know `docker stack`, so I have no idea what the differences between these are.

Reply:
The command docker stack ls returns the list of stacks in place. 

```
NAME                SERVICES            ORCHESTRATOR        NAMESPACE
grafana             2                   Kubernetes          default
kibana              1                   Kubernetes          default                     
```

We can keep the ls for now, but I'm think tasks is really useful too

The `name:` option is ommited when returning the list of stacks but when asking for a specific one (tasks) with the command `docker stack ps grafana`, it returns something like:

```
ID                  NAME                                IMAGE                         NODE                DESIRED STATE       CURRENT STATE            ERROR               PORTS
ef45cb00-3c3        grafana_grafana-0                   grafana/grafana               docker-desktop      Running             Running 22 minutes ago                       *:0->3000/tcp
0809163b-065        grafana_graphite-5b5b99dd5f-td4pm   graphiteapp/graphite-statsd   docker-desktop      Running             Running 22 minutes ago                       *:0->80/tcp,*:0->2003/tcp,*:0->2004/tcp
```

Reply:
So you have two completely different outputs, depending on what `action` is. I think it makes more sense to have different modules for this, than having one module that can do both.

How about having `docker_stack_info`, and `docker_stack_task_info`?

Also, why not using a similar interface as the other `docker_*_info` modules for similar objects? There's `docker_image_info` for returning information on multiple images, and there's `docker_container_info`, `docker_swarm_service_info` etc. on returning information on a single object.

Reply:
I love this idea. As soon as we resolve the other discussions I will split the module in two.